### PR TITLE
Fixes #30787 - upgrade js vendor to 8.3.0

### PR DESCRIPTION
**Pull request by MariaAga**
Fixes #30787 - upgrade to react-diff-view-2.x 
from 1.x to 2.x there were some breaking changes so the component should be refactored.
Needs https://github.com/theforeman/foreman-js/pull/217
Depends on https://github.com/theforeman/foreman/pull/7990 that has code adjustments for 7.0.0

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
@MariaAga what is the status, I believe the `foreman-js` PR is waiting for this to be ready? :thinking:

**Comment by MariaAga** [comment]
@ezr-ondrej I was waiting for Avi to do the merge as I'm not sure how to merge the foreman-js pr so it will be a breaking change :sweat_smile:

**Comment by MariaAga** [comment]
This PR is supporting the upgrade to react-diff-view-2.x that happens in vendor-core 8.0.0

Depends on https://github.com/theforeman/foreman/pull/7990 that has code adjustments for 7.0.0

**Comment by MariaAga** [comment]
No longer blocked

**Comment by tbrisker** [comment]
Packaging currently has vendor 6.0 and needs to be updated.

**Comment by MariaAga** [comment]
This is not an urgent update, and is here mostly so when the update happens the diffView won't be broken

**Comment by MariaAga** [comment]
Updating `@theforeman/vendor` to 7 causes overlayTrigger to stop working. This should not be merged yet. Looks like the css class "Open" isn't being added to dropdown for example

**Comment by sharvit** [comment]
Seems like the react upgrade to v17 cause the overlay issue.

1. I reverted and released v7.2.10 to npm so v7 should work now.
2. Created https://github.com/theforeman/foreman-js/pull/249 to master to fix v8

**Comment by MariaAga** [comment]
Thanks Avi, using 8.0.4 now, which has the fix for overlays

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "create-react-component": "yo react-domain"
   },
   "dependencies": {
-    "@theforeman/vendor": "^6.0.0",
+    "@theforeman/vendor": "^8.0.4",
```
Comment (xprazak2) [inline package.json]: Can we make it 8.3.0?

**Comment by xprazak2** [comment]
packaging: https://github.com/theforeman/foreman-packaging/pull/6290

File: package.json (hunk 1 above)
Comment (MariaAga) [reply package.json]: Yup

Comment (xprazak2) [reply package.json]: :+1:

**Review by evgeni — APPROVED** [review]
packaging :ghost: ACK on `package.json`

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**